src/nvim/ui_compositor.c

```diff
@@ -706,16 +706,21 @@ static void ui_comp_grid_resize(UI *ui, Integer grid, Integer width, Integer hei
 
 static void ui_comp_event(UI *ui, char *name, Array args)
 {
-  Error err = ERROR_INIT;
   UIEventCallback *event_cb;
   bool handled = false;
-
+  Error initerr = ERROR_INIT;
   map_foreach_value(&ui_event_cbs, event_cb, {
+    Error err = initerr;
     Object res = nlua_call_ref(event_cb->cb, name, args, false, &err);
     if (res.type == kObjectTypeBoolean && res.data.boolean == true) {
       handled = true;
     }
+    if (err.type != kErrorTypeNone) {
+      ELOG("Error while executing ui_comp_event callback: %s", err.msg);
+    }
+    api_clear_error(&err);
   })
+  api_clear_error(&initerr);
```

Comment:
`initerr.type` is always `kErrorTypeNone` so this is not needed.
```suggestion
```

Reply:
Thanks, made the update.

Interesting to me that that is how it works. The [NULL declaration for](https://github.com/andrewwillette/neovim-willette/blob/2dd8befde1ace73d84edae95c25d807bec0e2d86/src/nvim/api/private/defs.h#L15) `.msg` makes sense that it wouldn't take up any memory, but I'd think `-1` for `kErrorTypeNone` ([link](https://github.com/andrewwillette/neovim-willette/blob/2dd8befde1ace73d84edae95c25d807bec0e2d86/src/nvim/api/private/defs.h#L28)) would. I trust you, you know more than I do here.

Reply:
`type` is stored on the stack and does not need to be freed.

Reply:
Thank you. I understand.